source/postprocess/visualization/boundary_velocity_residual.cc

```diff
@@ -0,0 +1,260 @@
+		    prm.declare_entry ("Use spherical unit vectors", "false",
+							   Patterns::Bool (),
+							   "Specify velocity as r, phi, and theta components "
+							   "instead of x, y, and z. Positive velocities point up, east, "
+							   "and north (in 3D) or out and clockwise (in 2D). "
+							   "This setting only makes sense for spherical geometries.");
+		    prm.declare_entry ("Use GPS data", "false",
+							   Patterns::Bool (),
+							   "Use GPS data for residual instead of the GPlates velocity data.");
+		    }
+		  prm.leave_subsection();
+    	  }
+    	  prm.leave_subsection();
+    	}
+    	prm.leave_subsection();
+	  }
+
+
+      template <int dim>
+      void
+	  BoundaryVelocityResidual<dim>::parse_parameters (ParameterHandler &prm)
+	  {
+	    prm.enter_subsection("Postprocess");
+		  prm.enter_subsection("Visualization");
+		  {
+			prm.enter_subsection("Surface properties");
+			{
+			// Get the path to the data files. If it contains a reference
+			// to $ASPECT_SOURCE_DIR, replace it by what CMake has given us
+			// as a #define
+			ascii_data_directory = Utilities::expand_ASPECT_SOURCE_DIR(prm.get ("Ascii data directory"));
+			ascii_data_file_name    = prm.get ("Ascii data file name");
+			ascii_scale_factor      = prm.get_double ("Scale factor");
+
+			gplates_data_directory = Utilities::expand_ASPECT_SOURCE_DIR(prm.get ("GPlates data directory"));
+			gplates_data_file_name    = prm.get ("GPlates data file name");
+			gplates_scale_factor      = prm.get_double ("Scale factor");
+
+			use_spherical_unit_vectors = prm.get_bool("Use spherical unit vectors");
+			use_gps_data = prm.get_bool("Use GPS data");
+
+			if (use_spherical_unit_vectors)
+			  AssertThrow (this->get_geometry_model().natural_coordinate_system() == Utilities::Coordinates::spherical,
+			   ExcMessage ("Spherical unit vectors should not be used "
+				"when geometry model is not spherical."));
+			}
+			prm.leave_subsection();
+		  }
+		  prm.leave_subsection();
+	    prm.leave_subsection();
+	  }
+
+
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(BoundaryVelocityResidual,
+                                                  "boundary velocity residual",
+												  "A visualization output object that generates output for the velocity"
+												  "residual at the top surface. The residual is computed at each point as "
+												  "the difference of the modeled velocities from the input velocities at"
+												  "the surface for each vector component. The user has an option to choose"
+												  "the input data as the GPS velocities in the ASCII data format with columns "
+												  "in the same format as described for the initial temperature or compositions"
+												  "or a velocity filed computed from the GPlates program. ")
```

Comment:
I would write this like this:

```suggestion
												  "the surface for each vector component. The user has an option to choose"
												  "the input data as ascii data files (e.g. GPS velocities) with columns "
												  "in the same format as described for the 'ascii data' boundary velocity plugin "
												  "or a velocity field computed from the GPlates program 
												  "as described in the "gplates" boundary velocity plugin.")
```